backend/services/user/src/index.ts

```diff
@@ -18,6 +19,7 @@ async function createEndpoints(router: express.Router) {
 
   // GET endpoints
   router.get("/users", getAllUsers);
+  router.get("/users/exists/:username", getIsUsernameExists);
```

Comment:
@teoyuqi Instead of `/users/exists/:username`, would it be better if we had `/users/:username`, similar to `getUserByEmail`? in that case, then your code can simply check for example whether any rows were returned

Reply:
nevermind, discussed and realised this would clash! marking as resolved